src/complex/Utilities/OStreamUtilities.cpp

```diff
@@ -299,7 +299,14 @@ class TupleWriter : public ITupleWriter
 
     for(size_t index = 0; index < m_NumComps; index++)
     {
-      outputStrm << m_DataArray.getName() << "_" << index;
+      if(index == 0)
+      {
+        outputStrm << m_DataArray.getName();
+      }
+      else
+      {
+        outputStrm << m_DataArray.getName() << "_" << index;
+      }
```

Comment:
The issue here is that if there are 2 components then the first will be "Foo" while the second will be "Foo_1". To be clear what should happen is if there is only 1 component then the output should be:
`Foo`
if there are multiple components then the output should be:
`Foo_1,Foo_2`
 
Apologies for not being clear enough.

Reply:
Oh ok lll get that patched

Reply:
Waiting on the patch to re-approve.

Reply:
I so screwed that up... It _should_ have been:

if there are multiple components then the output should be:
```
Foo_0,Foo_1,Foo_2....
```

Reply:
I just pushed the fix but that will screw up the complex_plugins PR.